### ISSUE TITLE
iOS: Crash while canceling RDAuthViewController

### ISSUE THREAD

**Issue by potatolicious** [rdio/api#118]
Hello Rdio,

We're seeing a common crash when canceling out of the RDAuthViewController. An example crash log is attached:

```
Date/Time:       2014-05-01T22:48:52Z
OS Version:      iPhone OS 7.1.1 (11D201)
Report Version:  104

Exception Type:  SIGABRT
Exception Codes: #0 at 0x397f71f0
Crashed Thread:  0

Application Specific Information:
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[0]'

Last Exception Backtrace:
0   CoreFoundation                       0x2eaabf0b __exceptionPreprocess + 131
1   libobjc.A.dylib                      0x39242ce7 objc_exception_throw + 36
2   CoreFoundation                       0x2e9e9cff -[__NSPlaceholderDictionary initWithObjects:forKeys:count:] + 532
3   CoreFoundation                       0x2e9e9ac3 +[NSDictionary dictionaryWithObjects:forKeys:count:] + 48
4   YouandMe                             0x0044c363 -[RDAuthViewController restoreDefaultHeaders] (RDAuthViewController.m:254)
5   YouandMe                             0x0044c629 -[RDAuthViewController cancel] (RDAuthViewController.m:291)
6   UIKit                                0x313106c7 -[UIApplication sendAction:to:from:forEvent:] + 88
7   UIKit                                0x31310663 -[UIApplication sendAction:toTarget:fromSender:forEvent:] + 36
8   UIKit                                0x31310633 -[UIControl sendAction:to:forEvent:] + 44
9   UIKit                                0x312fbd7b -[UIControl _sendActionsForEvents:withEvent:] + 372
10  UIKit                                0x3131007b -[UIControl touchesEnded:withEvent:] + 592
11  UIKit                                0x3130fd4d -[UIWindow _sendTouchesForEvent:] + 526
12  UIKit                                0x3130aca7 -[UIWindow sendEvent:] + 756
13  UIKit                                0x312dfe75 -[UIApplication sendEvent:] + 194
14  UIKit                                0x312de541 _UIApplicationHandleEventQueue + 7118
15  CoreFoundation                       0x2ea76fef __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 12
16  CoreFoundation                       0x2ea764b7 __CFRunLoopDoSources0 + 204
17  CoreFoundation                       0x2ea74ca7 __CFRunLoopRun + 628
18  CoreFoundation                       0x2e9df769 CFRunLoopRunSpecific + 522
19  CoreFoundation                       0x2e9df54b CFRunLoopRunInMode + 104
20  GraphicsServices                     0x3394c6d3 GSEventRunModal + 136
21  UIKit                                0x3133e891 UIApplicationMain + 1134
22  YouandMe                             0x0010df1b main (main.m:16)
23  libdyld.dylib                        0x39740ab7 start + 0
```

It doesn't look like any code related to my app and seems self-contained to the Rdio SDK. If this is indeed on Rdio's end, is it possible to prioritize a fix? This is by far the largest source of crashes for our app overall.

**Comment by rknLA** [comment]
Hi @potatolicious!

This does look like it's something that's on our end.

Random, but actually related, question: Are you doing anything to the HTTP headers inside your app?

Thanks!
Kevin

**Comment by potatolicious** [comment]
Indeed - our backend is configured to expect a few custom header fields. I don't believe they apply to any network requests not from our AFHTTPRequestOperationManager subclasses though.

**Comment by rknLA** [comment]
Interesting.  I'm guessing that's semi-related.  Want to shoot over an email to developersupport@rd.io with the additional headers you're sending over so that we can test?

**Comment by dasevilla** [comment]
Fixed in v1.3.7

http://blog.rdio.com/developers/2014/05/ios-sdk-v137.html
